packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, SlotProvider, useFocusableRef, useResizeObserver, useStyleProps} from '@react-spectrum/utils';
+import {DOMProps, FocusableRef, SpectrumLabelableProps, StyleProps} from '@react-types/shared';
+import {Field} from '@react-spectrum/label';
+import React, {ReactNode, useCallback, useLayoutEffect, useRef, useState} from 'react';
+import styles from '@adobe/spectrum-css-temp/components/searchwithin/vars.css';
+import {useLabel} from '@react-aria/label';
+import {useProvider, useProviderProps} from '@react-spectrum/provider';
+
+export interface SpectrumSearchWithinProps extends SpectrumLabelableProps, DOMProps, StyleProps {
+  children: ReactNode,
+  /** Whether the children should be disabled. Propagated to both children. */
+  isDisabled?: boolean
+}
+
+function SearchWithin(props: SpectrumSearchWithinProps, ref: FocusableRef<HTMLElement>) {
+  props = useProviderProps(props);
+  let {styleProps} = useStyleProps(props);
+  let {labelProps} = useLabel(props);
+  let {
+    children,
+    isDisabled,
+    isRequired
+  } = props;
+
+  let inputRef = useRef<HTMLInputElement | HTMLTextAreaElement>();
+  let domRef = useFocusableRef(ref, inputRef);
+
+  // Measure the width of the field to inform the width of the menu.
+  let [menuWidth, setMenuWidth] = useState(null);
+  let {scale} = useProvider();
+
+  let onResize = useCallback(() => {
+    if (domRef.current) {
+      let width = domRef.current.offsetWidth;
+      setMenuWidth(width);
+    }
+  }, [domRef]);
+
+  useResizeObserver({
+    ref: domRef,
+    onResize: onResize
+  });
+
+  useLayoutEffect(onResize, [scale, onResize]);
+
+  let defaultSlotValues = {isDisabled, isRequired, label: undefined, isQuiet: false, 'aria-labelledby': labelProps.id};
```

Comment:
I don't think specifying label as undefined will prevent label from being set on the individual components?
I assume that isQuiet false is set for the same reason?
Can you make a chromatic story that has these cases where label/isQuiet is set on the individual children vs the SearchWithin?

Reply:
Good catch. Is there a proper way to override these? This was requested in the original issue (#1342).

Reply:
Well, we're going to need to figure out how we want to do this in a consistent way. Generally slots shouldn't override the props passed in directly on a child and it'd be breaking to start doing that now.
Instead, I think we'll need a new object inside of the slots context called 'override' or 'force' or something...
it could be used like so:
```
let slots = {
    searchfield: {UNSAFE_className: searchFieldClassName},
    picker: {UNSAFE_className: pickerClassName},
    overrides: {
      searchfield: {UNSAFE_className: searchFieldClassName, isQuiet: !!props.isQuiet, label: undefined},
      picker: {UNSAFE_className: pickerClassName, menuWidth, align: 'end', isQuiet: !!props.isQuiet, label: undefined},
    }
}
```

logic in https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/utils/src/Slots.tsx#L35
```
export function SlotProvider(props) {
  let parentSlots = useContext(SlotContext) || {};
  let {slots = {}, children} = props;
  // take overrides out of parentSlots and slots before using either object
  // do not use parentSlots.overrides??

  // Merge props for each slot from parent context and props
  let value = useMemo(() => 
    Object.keys(parentSlots)
      .concat(Object.keys(slots))
      .reduce((o, p) => ({
        ...o,
        [p]: mergeProps(
          parentSlots[p] || {},
          slots[p] || {}
          //, slots.overrides, this may still need something because it might now cause undefined to override a non-undefined value
       )}), {})
      , [parentSlots, slots]);

  return (
    <SlotContext.Provider value={value}>
      {children}
    </SlotContext.Provider>
  );
}
```

Reply:
A more unsavory option would be to do something w/ clone element and React.children.map (bleh) but the slots override seems like something we should do anyways

Reply:
ok, this is where we are testing the change of making slots override component props https://github.com/adobe/react-spectrum/pull/2073

for something like disabled, if isDisabled is specified on the SearchWithin, then it should take precedence (be passed on slots)
if it's not specified on SearchWithin, it should be passed on the provider context (already happens, really just need to remove it from the slots context for this case), so if it's specified on the scopes button, then it should apply (we don't want a prop on SearchWithin like 'scopesDisabled')

for something like isQuiet, this must be the same across both components, so we'd need always pass this value on the slots context

you can make the same change I did in my PR in order to try it out here